### PR TITLE
OKTA-364172: reload playground when mock data and configs are changed

### PR DESCRIPTION
**Pull request by lesterchoi-okta**
## Description:
Allows Okta SIW developers to see changes to playground configs reflected automatically without the need to manually reload the browser.

* uses `nodemon` and webpack `devServer.proxy` to restart mock server when files in `/playground` change.

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Reviewers:

@haishengwu-okta 

### Issue:

- [OKTA-364172](https://oktainc.atlassian.net/browse/OKTA-364172)

### CODE REVIEW

File: webpack.playground.config.js
Comment (lesterchoi-okta) [inline webpack.playground.config.js:57]: This comment already explains exactly why the changes weren't being detected.
The added helper function helps set the paths explicitly.
Alternatively, we could hardcode the paths if they don't change frequently.

File: helpers.js
Comment (lesterchoi-okta) [inline helpers.js:11]: Should move to a separate package or module and be tested.
Checked using `console.time`; takes ~4ms on `playground` directory.

**Review by lesterchoi-okta — COMMENTED** [review]
Open to suggestions on implementation strategy.
1. Hardcode paths
2. Move helper to package

File: webpack.playground.config.js
Comment (haishengwu-okta) [reply webpack.playground.config.js:57]: wow... this is simply. 🤦 
I think we can hard code the path, which are not changed frequently.

Comment (lesterchoi-okta) [reply webpack.playground.config.js:57]: Will do. Did you confirm it's reloading the way you want?

**Comment by pradeepdewda-okta** [comment]
@lesterchoi-okta wow thanks for doing this.. very helpful. Was on mind for a while.

File: webpack.playground.config.js
Comment (haishengwu-okta) [reply webpack.playground.config.js:54]: Did you verify if the mock data has been reflected after webpack reload?

Comment (lesterchoi-okta) [reply webpack.playground.config.js:54]: @haishengwu-okta Oops. Missed your edit. I just pushed the hardcoded version.
I can revert if you think the helper version is better, but I'd say the hardcoded paths makes it clear that subdirectories need to be added explicitly. If anything I should add a note to that section explaining that devs should update the webpack config with new paths if they're adding new folders.

Comment (lesterchoi-okta) [reply webpack.playground.config.js:54]: > Did you verify if the mock data has been reflected after webpack reload?

Yes, but I'll check again.

File: webpack.playground.config.js
Comment (pradeepdewda-okta) [inline webpack.playground.config.js:57]: this is too verbose.. I like helper better. we can add comment for explanation.

File: webpack.playground.config.js
Comment (haishengwu-okta) [reply webpack.playground.config.js:54]: looks like it will not reload the mock server

Comment (lesterchoi-okta) [reply webpack.playground.config.js:54]: IMHO the real issue is that these changes to `responseConfig.json` are tracked in git.

The **better** setup would be to have a sensible/common default for `responseConfig.json`, then have untracked "local" (i.e., personal or team) variations so devs aren't clobbering each other's test flows.

The **ideal** setup might be if it's a completely dynamic response that either...
1. uses magic test values to trigger specific responses (documentation required of course)
2. allows you to configure the responses via CLI (wishful)

Comment (lesterchoi-okta) [reply webpack.playground.config.js:54]: > looks like it will not reload the mock server

Looking into it

Comment (haishengwu-okta) [reply webpack.playground.config.js:54]: yeah, that's kind of real issue and #2 sounds good to me (CLI, or Web). 
but I'd assume it would take extra effort.
The ultimate goal is to make dev easier to use different set up of mocks.

#1 would require set up too many combinations.

Comment (lesterchoi-okta) [reply webpack.playground.config.js:54]: > looks like it will not reload the mock server

Okay. Yes, definitely jumped the gun on this (I must've restarted the dev-server manually without realizing). The routes are registered in the `before` hook in webpack-dev-server which is run once at the start, and not triggered for re-compilations. I'll investigate ways to make it happen on file changes.

File: webpack.playground.config.js
Comment (lesterchoi-okta) [reply webpack.playground.config.js:57]: @pradeepdewda-okta turns out this won't update the mock server anyway.
I'm going to cancel this PR and re-submit with a fix.

**Review by lesterchoi-okta — COMMENTED** [review]
Any thoughts on what, if any, updates I should make on `/playground/README.md`?

**Review by haishengwu-okta — APPROVED** [review]

**Comment by haishengwu-okta** [comment]
this is partially feature/partially bug fixing. so didn't think of change to the readme.
but if you find anything can improve for the readme, feel free to open PR.

**Comment by lesterchoi-okta** [comment]
@pradeepdewda-okta Passes [on Bacon](https://bacon-go.aue1e.saasure.net/commits?artifact=okta-signin-widget&sha=16b7143076896a3b40a4f828831e71b5b8ff0370)

I think this just needs "code owner review from okta/devex-engineering"

NOTE: Travis CI failure changes from run to run, guess there are some flaky tests?

**Pull request closed without merging**
